opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/ScopeManagerShim.java

```diff
@@ -29,8 +29,17 @@ public ScopeManagerShim(Tracer tracer) {
   }
 
   @Override
+  @SuppressWarnings("ReturnMissingNullable")
   public Span activeSpan() {
-    return new SpanShim(tracer.getCurrentSpan());
+    // As OpenTracing simply returns null when no active instance is available,
+    // we need to do an explicit check against DefaultSpan,
+    // which is used in OpenTelemetry for this very case.
+    io.opentelemetry.trace.Span span = tracer.getCurrentSpan();
+    if (span instanceof io.opentelemetry.trace.DefaultSpan) {
```

Comment:
I think we will use DefaultSpan for propagation of the SpanContext as well. The default Span when no Span in the context is DefaulSpan.getInvalid();

This is not right now the case, but will become soon, so you may simply compare with that from the beginning.

Reply:
> This is not right now the case, but will become soon, so you may simply compare with that from the beginning.

Correct, I was trying to be on the safe side ;) So lets change it the comparison to be against `DefaultSpan.getInvalid()` and lets update it in the future in case of any change.